File: PartyTime/src/main/java/com/pasciak/partytime/entities/Event.java
package com.pasciak.partytime.entities;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Objects;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;

@Entity
public class Event {

	@Id
	@Column(name = "id")
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	private String title;

	@Column(name = "lat")
	private Double lat;

	@Column(name = "lng")
	private Double lng;

	@Column(name = "datetime")
	private LocalDateTime dateTime;

	// @JsonIgnore
	@ManyToOne
	@JoinColumn(name = "user_id")
	private User user;

	public List<EventInvite> getEventInvites() {
		return eventInvites;
	}

	public void setEventInvites(List<EventInvite> eventInvites) {
		this.eventInvites = eventInvites;
	}

//	@JsonIgnore
	@JsonIgnoreProperties({ "event" })
	@OneToMany(mappedBy = "event")
	List<EventInvite> eventInvites;

	public Event() {
		super();
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public Double getLat() {
		return lat;
	}

	public void setLat(Double lat) {
		this.lat = lat;
	}

	public Double getLng() {
		return lng;
	}

	public void setLng(Double lng) {
		this.lng = lng;
	}

	public LocalDateTime getDateTime() {
		return dateTime;
	}

	public void setDateTime(LocalDateTime datetime) {
		this.dateTime = datetime;
	}

	public String getTitle() {
		return title;
	}

	public void setTitle(String title) {
		this.title = title;
	}

	@Override
	public int hashCode() {
		return Objects.hash(id);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Event other = (Event) obj;
		return Objects.equals(id, other.id);
	}

	@Override
	public String toString() {
		return "Event [id=" + id + ", lat=" + lat + ", lng=" + lng + ", dateTime=" + dateTime + "]";
	}

	public User getUser() {
		return user;
	}

	public void setUser(User user) {
		this.user = user;
	}

}

File: ngPartyTime/src/app/models/event.ts
import { EventInvite } from './event-invite';
import { User } from './user';

export class Event {
  id: number;
  lat?: number | null;
  lng?: number | null;
  title: string;
  userId: number;
  dateTime?: Date | null;
  user: User | null;
  eventInvites: EventInvite[] | null;

  constructor(
    id: number = 0,
    lat: number | null = 0,
    lng: number | null = 0,
    title: string = '',
    userId: number = 0,
    dateTime?: Date | null,
    user: User | null = null,
    eventInvites: EventInvite[] | null = null
  ) {
    this.id = id ?? 0;
    this.lat = lat ?? null;
    this.lng = lng ?? null;
    this.title = title ?? 'Unknown Title Of Event';
    this.userId = userId ?? 0;
    this.dateTime = dateTime ? new Date(dateTime) : null;
    this.user = user ?? null;
    this.eventInvites = eventInvites ?? null;
  }
}
